extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php

```diff
@@ -253,19 +221,21 @@ static private function getConfigForWiki( $cityId ){
 	 * it as a parsed object.
 	 */
 	static private function getAllAppConfigs(){
-		wfProfileIn(__METHOD__);
 		// Pull in the app-configuration (has data for all apps)
 		$appConfig = [];
 		$memcKey = wfMemcKey( static::$CACHE_KEY, static::$CACHE_KEY_VERSION );
 		$response = F::app()->wg->memc->get( $memcKey );
 		if ( empty( $response ) ){
-			$req = MWHttpRequest::factory( self::APP_CONFIG_SERVICE_URL, array( 'noProxy' => true ) );
+			$req = MWHttpRequest::factory( static::APP_CONFIG_SERVICE_URL, array( 'noProxy' => true ) );
 			$status = $req->execute();
 			if( $status->isOK() ) {
 				$response = $req->getContent();
 				if ( empty( $response ) ) {
-					wfProfileOut( __METHOD__ );
-					throw new EmptyResponseException( self::APP_CONFIG_SERVICE_URL );
+					// Request failed (app config service failure). Do not cache this response.
+					$this->error(__METHOD__ . ' app config service failure.', [
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Remove this use of "$this". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2014)